ecl/hql/hqlopt.cpp

```diff
@@ -2494,6 +2494,34 @@ IHqlExpression * CTreeOptimizer::doCreateTransformed(IHqlExpression * transforme
                 node_operator childOp = child->getOperator();
                 switch (childOp)
                 {
+                case no_selectnth:
+                    {
+                        IHqlExpression * child0 = child->queryChild(0);
+                        if (child0->getOperator()==no_newaggregate)
+                        {
+                            if (isSimpleCountExistsAggregate(child0, false, true))
+                            {
+                                IHqlExpression * selectnth = child->queryChild(1);
+                                IHqlExpression * dataset = child0->queryChild(0);
+                                if (!isShared(dataset) && getIntValue(selectnth,-1)==1 )
+                                {
+                                    if (dataset->getOperator() == no_addfiles)
+                                    {
+                                        OwnedHqlExpr newLeftDS = replaceDataset(child, dataset, dataset->queryChild(0));
+                                        IHqlExpression * field = newLeftDS->queryRecord()->queryChild(0);
```

Comment:
Should be ok to use

field = transformed->queryChild(1);

and then use the same expression for all the createNewSelectExpr() calls.